x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -190,11 +190,41 @@ export function trainedModelsRoutes(
             // we don't need to fill kibana's log with these messages.
             mlLog.debug(e);
           }
+          const enabledFeatures = getEnabledFeatures();
+          try {
+            if (enabledFeatures.dfa) {
+              const jobIds = result.map((model) => {
+                let id = model.metadata?.analytics_config?.id;
+                if (id) {
+                  id = `${id}*`;
+                }
+                return id;
+              });
+              const filteredJobIds = jobIds.filter((id) => id !== undefined);
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+              const { data_frame_analytics: jobs } = await mlClient.getDataFrameAnalytics({
+                id: filteredJobIds.join(','),
```

Comment:
The `map`, `filter` and `join` above can all be combined into a single `reduce` to create the comma separated string.

Reply:
Updated in 50ffcfdd594f6cedb13acedc195b6cd00bac950d